examples/dann.py

```diff
@@ -0,0 +1,307 @@
+        net = self._build_feature_extractor(main_input)
+        self.grl = GradientReversal()
+        branch = self.grl(net, hp_lambda)
+        branch = Dense(128, activation='relu')(branch)
+        branch = Dropout(0.1)(branch)
+        branch = Dense(2, activation='softmax', name='domain_output')(branch)
+
+        # When building DANN model, route first half of batch (source examples)
+        # to domain classifier, and route full batch (half source, half target)
+        # to the domain classifier.
+        net = Lambda(lambda x: K.switch(K.learning_phase(), x[:int(batch_size / 2), :], x, lazy=True),
+                     output_shape=lambda x: ((batch_size / 2,) + x[1:]))(net)
+
+        net = self._build_classifier(net)
+        model = Model(input=main_input, output=[branch, net])
+        if plot_model:
+            plot(model, show_shapes=True)
+        model.compile(loss={'classifier_output': 'categorical_crossentropy',
+                      'domain_output': 'categorical_crossentropy'},
+                      optimizer=self.opt, metrics=['accuracy'])
+        return model
+
+    def build_tsne_model(self, main_input):
+        '''Create model to output intermediate layer
+        activations to visualize domain invariant features'''
+        tsne_model = Model(input=main_input,
+                           output=self.domain_invariant_features)
+        return tsne_model
+
+
+main_input = Input(shape=(3, img_rows, img_cols), name='main_input')
+
+builder = DANNBuilder()
+src_model = builder.build_source_model(main_input)
+src_vis = builder.build_tsne_model(main_input)
+
+hp_lambda = K.variable(1.0)
+dann_model = builder.build_dann_model(main_input, hp_lambda)
+dann_vis = builder.build_tsne_model(main_input)
+print('Training source only model')
+src_model.fit(X_train, y_train, batch_size=64, nb_epoch=10, verbose=1,
+              validation_data=(X_test, y_test))
+print('Evaluating target samples on source-only model')
+print('Accuracy: ', src_model.evaluate(XT_test, y_test)[1])
+
+# Broken out training loop for a DANN model.
+src_index_arr = np.arange(X_train.shape[0])
+target_index_arr = np.arange(XT_train.shape[0])
+
+batches_per_epoch = len(X_train) / batch_size
+num_steps = nb_epoch * batches_per_epoch
+j = 0
+
+print('Training DANN model')
+
+for i in range(nb_epoch):
+
+    batches = make_batches(X_train.shape[0], batch_size / 2)
+    target_batches = make_batches(XT_train.shape[0], batch_size / 2)
+
+    src_gen = batch_gen(batches, src_index_arr, X_train, y_train)
+    target_gen = batch_gen(target_batches, target_index_arr, XT_train, None)
+
+    losses = list()
+    acc = list()
+
+    print('Epoch ', i)
+
+    for (xb, yb) in src_gen:
+
+        # Update learning rate and gradient multiplier as described in
+        # the paper.
+        p = float(j) / num_steps
+        l = 2. / (1. + np.exp(-10. * p)) - 1
+        lr = 0.01 / (1. + 10 * p)**0.75
+        hp_lambda = l
+        builder.opt.lr = lr
```

Comment:
I'm not sure, but it seems, that here you just override `hp_lambda` and `builder.opt.lr` to become float numbers, but don't change the real gradient multiplier and learning rate in the graph. I believe, that you need to use `K.set_value()` as it is done in [LearningRateScheduler callback](https://github.com/fchollet/keras/blob/master/keras/callbacks.py#L424)

Reply:
Thanks for pointing this! Yes, this changes are not affected learning rate and hp_lambda. I tried with Keras 2 (Tensorflow). In order to make changes to parameters hp_lambda and learning rate, I need to use K.set_value.